src/runtime/rocm/rocm_device_api.cc

```diff
@@ -0,0 +1,142 @@
+/*!
+ *  Copyright (c) 2017 by Contributors
+ * \file rocm_device_api.cc
+ * \brief GPU specific API
+ */
+#include <tvm/runtime/config.h>
+#include <tvm/runtime/device_api.h>
+
+#if TVM_ROCM_RUNTIME
+#include <dmlc/logging.h>
+#include <dmlc/thread_local.h>
+#include <tvm/runtime/registry.h>
+#include <hip/hip_runtime_api.h>
+#include "./rocm_common.h"
+
+namespace tvm {
+namespace runtime {
+
+class ROCMDeviceAPI final : public DeviceAPI {
+ public:
+  void SetDevice(TVMContext ctx) final {
+    ROCM_CALL(hipSetDevice(ctx.device_id));
+  }
+  void GetAttr(TVMContext ctx, DeviceAttrKind kind, TVMRetValue* rv) final {
+    int value = 0;
+    switch (kind) {
+      case kExist:
+        value = 1024;
```

Comment:
This is wrong, check the CUDA counter part
https://github.com/dmlc/tvm/blob/master/src/runtime/cuda/cuda_device_api.cc#L28

It checks if the query succeeded, to check if the corresponding GPU device exists.